tools/src/main/java/com/nedap/archie/rmobjectvalidator/RMObjectValidator.java

```diff
@@ -130,6 +148,34 @@ else if (cobject instanceof CPrimitiveObject) {
         return result;
     }
 
+    private void validateArchetypeSlot(List<RMObjectWithPath> rmObjects, String path, CObject cobject, List<RMObjectValidationMessage> result) {
+        ArchetypeSlot slot = (ArchetypeSlot) cobject;
+        for (RMObjectWithPath objectWithPath : rmObjects) {
+
+            Object object = objectWithPath.getObject();
+
+            String archetypeId =  metaModel.getSelectedModel().getArchetypeIdFromArchetypedRmObject(object);
+            if(archetypeId != null) {
```

Comment:
You're right, that is not correct. It should indeed still validate the RM Objects. I will change that. In addition, it should probably add another error that an archetype id should be set here?

Another possible addition is that the node id can be the archetype id. I think that is really ugly for several reasons, but it is how ADL 1.4 did it because node ids for archetype slots weren't mandatory. I'll add that as well.